tools/python/util/convert_onnx_models_to_ort.py

```diff
@@ -0,0 +1,206 @@
+# Licensed under the MIT License.
+
+import argparse
+import os
+import pathlib
+import typing
+
+import onnxruntime as ort
+from .ort_format_model import create_config_from_models
+
+
+def _path_match_suffix_ignore_case(path: typing.Union[pathlib.Path, str], suffix: str):
+    if not isinstance(path, str):
+        path = str(path)
+    return path.casefold().endswith(suffix.casefold())
+
+
+def _onnx_model_path_to_ort_model_path(onnx_model_path: pathlib.Path, optimization_level_str: str):
+    assert onnx_model_path.is_file() and _path_match_suffix_ignore_case(onnx_model_path, ".onnx")
+    return onnx_model_path.with_suffix(".{}.ort".format(optimization_level_str))
+
+
+def _create_config_file_from_ort_models(onnx_model_path_or_dir: pathlib.Path, enable_type_reduction: bool):
+    if onnx_model_path_or_dir.is_dir():
+        # model directory
+        model_path_or_dir = onnx_model_path_or_dir
+        config_path = None  # default path in model directory
+    else:
+        # single model
+        model_path_or_dir = _onnx_model_path_to_ort_model_path(onnx_model_path_or_dir)
+        config_suffix = ".{}".format(
+            'required_operators_and_types.config' if enable_type_reduction else 'required_operators.config')
+        config_path = model_path_or_dir.with_suffix(config_suffix)
+
+    create_config_from_models(model_path_or_dir=str(model_path_or_dir),
+                              output_file=str(config_path) if config_path is not None else None,
+                              enable_type_reduction=enable_type_reduction)
+
+
+def _create_session_options(optimization_level: ort.GraphOptimizationLevel,
+                            output_model_path: pathlib.Path,
+                            custom_op_library: pathlib.Path):
+    so = ort.SessionOptions()
+    so.optimized_model_filepath = str(output_model_path)
+    so.graph_optimization_level = optimization_level
+
+    if custom_op_library:
+        so.register_custom_ops_library(str(custom_op_library))
+
+    return so
+
+
+def _convert(model_path_or_dir: pathlib.Path, optimization_level_str: str, use_nnapi: bool,
+             custom_op_library: pathlib.Path, create_optimized_onnx_model: bool):
+
+    optimization_level = _get_optimization_level(optimization_level_str)
+
+    models = []
+    if model_path_or_dir.is_file() and _path_match_suffix_ignore_case(model_path_or_dir, ".onnx"):
+        models.append(model_path_or_dir)
+    elif model_path_or_dir.is_dir():
+        for root, _, files in os.walk(model_path_or_dir):
+            for file in files:
+                if _path_match_suffix_ignore_case(file, ".onnx"):
+                    models.append(pathlib.Path(root, file))
+
+    if len(models) == 0:
+        raise ValueError("No .onnx files were found in '{}'".format(model_path_or_dir))
+
+    providers = ['CPUExecutionProvider']
+    if use_nnapi:
+        # providers are priority based, so register NNAPI first
+        providers.insert(0, 'NnapiExecutionProvider')
+
```

Comment:
The package may not build with NNAPI, can we check if NNAPI is part of get_available_providers, display a warning and move optimization level to 1 here?

Reply:
Will add check. 

I think the optimization level is fine though - if someone specifes 'use_nnapi' when running this script we do the special handling to preserve all nodes NNAPI can run.